pkg/gateway/model_build_listener.go

```diff
@@ -19,38 +17,39 @@ const (
 	awsCustomCertARN = "application-networking.k8s.aws/certificate-arn"
 )
 
-func (t *latticeServiceModelBuildTask) extractListenerInfo(ctx context.Context, parentRef gateway_api.ParentReference) (int64, string, string, error) {
-
-	var protocol gateway_api.ProtocolType = gateway_api.HTTPProtocolType
+func (t *latticeServiceModelBuildTask) extractListenerInfo(
+	ctx context.Context,
+	parentRef gateway_api.ParentReference,
+) (int64, string, string, error) {
```

Comment:
need to wrap return arguments into struct

Reply:
Would like to avoid changing too much now. While 3 return values is more than what we'd like, it's also not too obtrusive to read at the moment.

Reply:
Multi-return purpose is not for results that are suppose to be used together, it's anti-pattern of usage. 

You wouldn't see returning more than 3 values in golang projects. Most of the time it's (result, error), (result, bool), and rare occasions with 3 values for example create input/output channels for socket - createConnection() (in, out, error), so in/out can be used separately, likely in different go-routines.

Since it's old code, we can change it after this PR.